cni/pkg/nodeagent/informers.go

```diff
@@ -83,6 +83,11 @@ func setupHandlers(ctx context.Context, kubeClient kube.Client, dataplane MeshDa
 }
 
 func (s *InformerHandlers) GetPodIfAmbient(podName, podNamespace string) (*corev1.Pod, error) {
+	// Both the CNI plugin and the informer check the K8S API to validate that the pod in
+	// question should be part of ambient, on "both ends".
+	// In the informer, wait for cache sync here to make sure we are fresh, to avoid issues
+	// where this cache is out of date with what the plugin might have seen when contacting the k8s API directly.
+	kube.WaitForCacheSync("informer", s.ctx.Done(), s.pods.HasSynced, s.namespaces.HasSynced)
```

Comment:
We should probably block in Start() before we serve, not here? Or if not, can you give some context on why we vary from the typical pattern here?

Reply:
As mentioned, I didn't realize that this was only useful to call once. Will remove.